### PR TITLE
feat(scripts): add todo detail registry for rich context preservation

### LINKED ISSUE #374: Todo items lose rich context — only brief description persisted to STATE.md
**Problem**

When users add todos via `/vbw:todo`, only a one-line bullet lands in `STATE.md ## Todos`. If the user provided substantial context — reproduction steps, file paths, stack traces, design intent, multi-sentence explanations — that detail is discarded. When the todo is later picked up via `/vbw:list-todos` and routed to `/vbw:fix` or `/vbw:vibe`, the executing agent sees only the brief description. The richer context that motivated the todo is gone.

Similarly, when `track-known-issues.sh promote-todos` promotes known issues into `STATE.md ## Todos`, the structured detail from `known-issues.json` (test name, file, error, times_seen) is flattened into a single-line bullet. The registry data exists but isn't bridged to the todo consumer.

**Root cause analysis**

`commands/todo.md` has no concept of extended detail storage — it writes a single formatted bullet to `STATE.md ## Todos` and stops. `scripts/list-todos.sh` parses those bullets into JSON but has no mechanism for supplementary context. There is no sidecar storage for todo-level detail.

The `known-issues.json` pattern already solves this for phase-scoped QA issues (brief reference in human-readable artifacts, structured detail in a machine-readable registry), but no equivalent exists for the project-level todo backlog.

**Affected components**

- `scripts/todo-details.sh` — **NEW** CRUD helper for the detail registry
- `commands/todo.md` — add conditional detail-capture step
- `commands/list-todos.md` — display detail on selection, pass ref to follow-up command
- `commands/fix.md` — conditional detail load at execution start
- `commands/vibe.md` — conditional detail load at Execute/Fix mode start
- `scripts/list-todos.sh` — parse `(ref:<hash>)` from todo lines, add `ref` field to JSON output
- `scripts/track-known-issues.sh` — update `promote-todos` to write detail entries
- `scripts/persist-state-after-ship.sh` — add GC call after STATE.md rebuild
- `scripts/bootstrap/bootstrap-state.sh` — initialize empty `todo-details.json`
- `testing/verify-init-todo.sh` — add contract checks

**Proposed solution**

A sidecar JSON registry (`.vbw-planning/todo-details.json`) that stores optional extended context per todo, keyed by a stable 8-char hash. `STATE.md` bullets remain brief (token-friendly for context compilation), but the detail file is loaded on-demand only when a specific todo is selected for action. This mirrors the existing `known-issues.json` pattern.

- **Write path**: `/vbw:todo` detects rich context (file paths, reproduction steps, stack traces, code references, error messages, design rationale) and stores it in the registry. `(ref:<hash>)` is appended to the STATE.md bullet. `promote-todos` also writes detail entries from known-issue data.
- **Read path**: `/vbw:list-todos` shows extended detail when a todo is selected. The ref is included in the follow-up command invocation so executing commands (`/vbw:fix`, `/vbw:vibe`) can load the detail as initial context.
- **Lifecycle**: Detail entries are GC'd on todo removal and milestone ship. Bootstrap initializes an empty registry.

**Acceptance criteria**

1. `scripts/todo-details.sh` exists with `add`, `get`, `remove`, `list`, and `gc` subcommands, all producing valid JSON output
2. `scripts/todo-details.sh add` upserts entries in `.vbw-planning/todo-details.json` using atomic write (tmp+mv)
3. `scripts/todo-details.sh get <hash>` returns the detail JSON for a given hash, or error status for missing entries
4. `scripts/todo-details.sh gc <state-path>` removes entries whose hash does not appear in `STATE.md ## Todos`
5. `scripts/list-todos.sh` parses `(ref:<hash>)` from todo lines and includes `ref` field in JSON output items
6. `commands/todo.md` captures extended detail when the user's input contains file paths, reproduction steps, stack traces, code references, error messages, or design rationale — appending `(ref:<hash>)` to the STATE.md bullet
7. `commands/list-todos.md` displays extended detail when a todo with a ref is selected, and includes the ref in the follow-up command suggestion
8. `commands/fix.md` and `commands/vibe.md` parse `(ref:<hash>)` from input, strip it before routing, and load detail for action-bearing modes only
9. `commands/fix.md` and `commands/vibe.md` gracefully continue without detail if the lookup fails
10. `scripts/track-known-issues.sh` `promote-todos` writes a detail entry for each promoted known issue
11. `scripts/bootstrap/bootstrap-state.sh` initializes empty `todo-details.json` alongside `STATE.md`
12. `scripts/persist-state-after-ship.sh` calls `todo-details.sh gc` after rebuilding STATE.md
13. BATS tests (`tests/todo-details.bats`) cover add/get/remove/gc subcommands
14. Contract tests (`testing/verify-init-todo.sh`) verify bootstrap creates `todo-details.json` and list-todos handles ref tags
15. Existing `tests/list-todos.bats` tests continue to pass (backward compatible)
16. `bash testing/run-all.sh` passes with zero failures

**Scope boundary**

NOT in scope:
- Auto-capture from conversation context (scanning prior conversation history for relevant context)
- Changes to `compile-context.sh` (detail is on-demand, not pre-compiled into agent context)
- Migration of existing todos (additive — only new todos with rich context get detail entries)
- Detail for todos added by commands other than `/vbw:todo` and `promote-todos`

**Alternatives considered**

1. **Inline detail in STATE.md** — rejected because it bloats the token footprint when `compile-context.sh` or agents read STATE.md. Sidecar keeps the human-readable artifact scannable.
2. **Line-number or text-matching keys** — rejected because they break on reordering, whitespace changes, or text edits. Hash-based keying is stable.
3. **Session memory for detail** — rejected because session memory doesn't survive across sessions. The detail must persist across the session boundary between "add todo" and "execute todo."

### PR DESCRIPTION
**Pull request by dpearson2699**
Fixes #374

## What

Introduces a sidecar JSON registry (`.vbw-planning/todo-details.json`) that stores extended context for todo items. When a todo contains rich information (file paths, reproduction steps, stack traces, code references, error messages, or design rationale), the detail is persisted separately and linked via a `(ref:<hash>)` suffix on the STATE.md bullet. This context is loaded on-demand when executing the todo, avoiding token bloat in compiled context while preserving the information across sessions.

### Files Modified

| File | Change |
|------|--------|
| `scripts/todo-details.sh` (NEW) | CRUD helper with 5 subcommands: `add`, `get`, `remove`, `list`, `gc`. Atomic writes via tmp+mv, fail-open design, 2000-char context cap. Stdin support via `-` sentinel for safe JSON piping. jq-based error_json for proper JSON escaping. |
| `tests/todo-details.bats` (NEW) | 18 BATS behavior tests covering all subcommands, edge cases, stdin piping, JSON escaping, malformed registry recovery, and legacy heading support. |
| `commands/todo.md` | Step 4 captures extended detail conditionally, computes 8-char shasum hash, stores via heredoc stdin pipe to `todo-details.sh add`, appends `(ref:<hash>)` to STATE.md bullet. |
| `commands/list-todos.md` | Step 6 loads detail on selection via `todo-details.sh get`. Step 8 strips metadata and includes ref in follow-up command for fix/vibe only (not debug/research which don't parse refs). Remove action cleans up detail entry. |
| `commands/fix.md` | Step 1 parses/strips `(ref:<hash>)` from arguments, loads detail for Dev task context. |
| `commands/vibe.md` | Input Parsing pre-parse extracts ref tag. Execute mode step 3 loads detail on-demand (deferred to action-bearing mode only). |
| `scripts/list-todos.sh` | `parse_todo_line()` extracts `(ref:<hash>)` as a field. JSON output includes `ref` field. Display stripping handles both ref and date tags in correct order. |
| `scripts/track-known-issues.sh` | `promote_todos()` computes ref hash, writes `(ref:<hash>)` to STATE.md, stores detail entries via session symlink. Only processes actually promoted items. Preserves existing ref tags during disposition updates. |
| `scripts/bootstrap/bootstrap-state.sh` | Initializes empty `todo-details.json` alongside STATE.md during bootstrap. |
| `scripts/persist-state-after-ship.sh` | Calls `todo-details.sh gc` after STATE.md rebuild to remove orphaned entries. |
| `scripts/verify-init-todo.sh` | Added BOOT-05/06 (bootstrap creates valid todo-details.json) and LIST-04/05 (ref parsing in list-todos.sh) contract checks. |

## Why

When adding a todo, users often provide rich context (file paths, stack traces, design rationale). The existing one-line todo format in STATE.md loses this information. When the todo is later picked up for execution, the executing agent has no context about what was originally described. This is especially problematic across sessions where conversation history is unavailable.

The sidecar registry pattern (detail stored in a separate JSON file, referenced by hash) preserves the information without inflating STATE.md or compiled context. Detail is loaded on-demand only when the todo is actually executed, keeping the token footprint minimal.

## How

The design uses content-addressable storage: each todo's detail is keyed by the first 8 characters of a shasum hash of the todo text. The hash appears as `(ref:<hash>)` at the end of the STATE.md bullet. Commands that execute todos (`/vbw:fix`, `/vbw:vibe`) parse this suffix, strip it from the routed arguments, and load the detail from the registry.

The registry lifecycle is: **create** (during `/vbw:todo` or `promote-todos`) → **read** (during `/vbw:fix`, `/vbw:vibe`, `/vbw:list-todos`) → **delete** (during `/vbw:list-todos` remove action or `gc` after ship). The `gc` subcommand removes entries whose hash no longer appears in STATE.md, preventing unbounded registry growth.

JSON detail is piped via heredoc stdin (avoiding shell-quoting hazards with apostrophes/special characters in user text). Error output uses jq for proper JSON escaping.

## Acceptance Criteria Verification

1. **todo-details.sh with 5 subcommands**: Satisfied — `scripts/todo-details.sh` dispatches `add`, `get`, `remove`, `list`, `gc` via case statement, all output valid JSON (including error paths via jq).
2. **add upserts with atomic write**: Satisfied — `write_registry()` uses `mktemp` + `mv`. `cmd_add()` supports both positional arg and stdin (`-` sentinel).
3. **get returns detail or error status**: Satisfied — returns `{"status":"ok",...}` or `{"status":"not_found",...}`. Invalid hash format returns `not_found` with properly escaped JSON.
4. **gc removes orphaned entries**: Satisfied — awk extracts refs from both `## Todos` and `### Pending Todos` headings, removes keys not found in STATE.md.
5. **list-todos.sh parses ref and includes ref field**: Satisfied — `parse_todo_line()` extracts ref via regex, JSON items include `ref` field (null when absent). Contract test LIST-04 verifies.
6. **todo.md captures extended detail conditionally**: Satisfied — Step 4 detects rich context triggers, computes hash, stores detail via heredoc stdin pipe, appends ref tag.
7. **list-todos.md displays detail and includes ref in follow-up**: Satisfied — Step 6 loads detail on selection. Step 8 strips metadata before building follow-up command, appends ref only for fix/vibe commands.
8. **fix.md and vibe.md parse ref for action-bearing modes**: Satisfied — fix.md Step 1 parses and strips. vibe.md defers detail loading to Execute mode Step 3 only.
9. **Graceful continue on failed lookup**: Satisfied — Both fix.md and vibe.md explicitly continue without detail on `not_found` or `error`.
10. **promote-todos writes detail entries**: Satisfied — Resolves `todo-details.sh` via session symlink, calls `add` for each promoted item with structured detail JSON.
11. **bootstrap initializes empty todo-details.json**: Satisfied — Creates `{"schema_version":1,"items":{}}` if file doesn't exist. Contract test BOOT-05/06 verifies.
12. **persist-state-after-ship calls gc**: Satisfied — Resolves plugin root via session symlink, calls `gc` with the freshly-generated STATE.md path. Fail-open.
13. **BATS tests cover subcommands**: Satisfied — 18 tests in `tests/todo-details.bats` covering add (4 + stdin), get (3), remove (2), list (2), gc (3 including legacy heading), error handling (2 + JSON escaping), stdin (1).
14. **Contract tests verify bootstrap and list-todos**: Satisfied — BOOT-05, BOOT-06, LIST-04, LIST-05 in `scripts/verify-init-todo.sh`.
15. **Existing list-todos.bats backward compatible**: Satisfied — New `ref` field added alongside existing fields; null for todos without ref tags. No existing assertions broken.
16. **run-all.sh passes**: Satisfied — 33/33 contract, 2588/2588 BATS, 1/1 lint.

## Testing

- [x] Load plugin locally with `claude --plugin-dir .`
- [x] `bash testing/run-all.sh` passes: 33/33 contract, 2588/2588 BATS, 1/1 lint
- [x] New BATS tests: 18 tests in `tests/todo-details.bats`
- [x] New contract tests: BOOT-05, BOOT-06, LIST-04, LIST-05
- [x] Existing tests unaffected

### Phase 3 — Primary QA (Claude Opus 4.6, qa-investigator): 3 rounds
- **Round 1**: 6 findings (1 high, 3 medium, 2 low) — all fixed in `6e0a206`
  - F-01 (high): track-known-issues.sh PLUGIN_ROOT unset → fixed to use session-link pattern
  - F-02 (medium): sed stripping order in list-todos.sh → swapped ref before date
  - F-03 (medium): disposition updates strip ref tags → preserve existing ref
  - F-04 (low): persist-state-after-ship.sh broken primary path → session-link only
  - F-05 (medium): missing LIST-04/05 contract tests → added
  - F-06 (low): vibe.md loads detail unconditionally → deferred to Execute mode
- **Round 2**: 3 findings (all low) — all fixed in `55759cd`
  - F-01: gc dual heading support → added `### Pending Todos` to awk pattern
  - F-02: hash validation inconsistency → added validation to get/remove
  - F-03: detail loop iterates non-promoted items → iterate only promoted items
- **Round 3**: 1 finding (low) — fixed in `94b13d4`
  - F-01: missing BATS test for gc legacy heading → added test

### Phase 3.5 — Cross-Model QA (GPT-5.4, qa-investigator-gpt-54): 3 rounds
- **Round 1**: 3 findings (1 high, 2 medium) — 1 fixed, 2 false positives in `c3f8959`
  - F-01 (high, false positive): Hash collision for duplicate summaries — by-design content-addressable behavior; identical todo text IS the same todo
  - F-02 (medium, fixed): list-todos.md follow-up command had duplicate ref tags → strip metadata before building command
  - F-03 (medium, false positive): Session symlink dependency — standard architectural pattern used by all scripts; scripts already fail gracefully
- **Round 2**: 2 findings (1 medium, 1 low) — both fixed in `1c70a30`
  - F-01 (medium): Shell quoting in todo.md breaks on apostrophes → heredoc stdin pipe pattern
  - F-02 (low): error_json() unescaped JSON interpolation → jq-based JSON escaping + fixed unvalidated hash interpolation in cmd_get/cmd_remove
- **Round 3**: 3 findings (1 medium, 2 low) — 1 fixed, 2 false positives in `b720681`
  - F-01 (medium, false positive): promote-todos session symlink only — same standard pattern as Round 1 F-03, function only runs within active Claude sessions
  - F-02 (low, false positive): persist-state-after-ship session symlink — same pattern, GC is best-effort
  - F-03 (low, fixed): list-todos ref appended to debug/research follow-up → only append for fix/vibe

### Totals
- **Phase 3**: 10 findings, 10 fixed, 0 false positives
- **Phase 3.5**: 8 findings, 4 fixed, 4 false positives
- **Grand total**: 18 findings across 6 rounds, 14 fixed with root-cause remediation, 4 classified as false positives

### CODE REVIEW

File: commands/list-todos.md [hunk 1]
```diff
@@ -63,10 +65,11 @@ allowed-tools: Read, Edit, Bash, AskUserQuestion
      ```text
      ✓ Todo picked up.
 
-     ➜ Run: /vbw:{command} {todo text}
+     ➜ Run: /vbw:{command} {cleaned text} (ref:HASH)
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply commands/list-todos.md]: Fixed in b778bee. Example now shows `[(ref:HASH) if fix/vibe and ref exists]` to reflect the conditional logic described in the instruction text.

File: scripts/track-known-issues.sh [hunk 2]
```diff
@@ -746,17 +747,28 @@ promote_todos() {
...
+
       if [ "$disposition" = "accepted-process-exception" ]; then
-        new_entries="${new_entries}- [KNOWN-ISSUE] ${test_name} (${file_path}): ${error_msg} — accepted as process-exception for this phase (phase ${phase_num}, seen ${times_seen}x)${source_ref} (added ${today})"$'\n'
+        new_entries="${new_entries}- [KNOWN-ISSUE] ${test_name} (${file_path}): ${error_msg} — accepted as process-exception for this phase (phase ${phase_num}, seen ${times_seen}x)${source_ref} (added ${today}) (ref:${ref_hash})"$'\n'
       else
-        new_entries="${new_entries}- [KNOWN-ISSUE] ${test_name} (${file_path}): ${error_msg} (phase ${phase_num}, seen ${times_seen}x)${source_ref} (added ${today})"$'\n'
+        new_entries="${new_entries}- [KNOWN-ISSUE] ${test_name} (${file_path}): ${error_msg} (phase ${phase_num}, seen ${times_seen}x)${source_ref} (added ${today}) (ref:${ref_hash})"$'\n'
       fi
+      # Collect detail for newly promoted items
+      promoted_details="${promoted_details}${ref_hash}"$'\x1f'"${test_name}"$'\x1f'"${file_path}"$'\x1f'"${error_msg}"$'\x1f'"${times_seen}"$'\x1f'"${source_artifact}"$'\x1f'"${disposition}"$'\n'
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/track-known-issues.sh]: Fixed in b778bee. Now captures `full_error_msg` before truncation and passes it through `promoted_details` to the detail JSON context field. The truncated `error_msg` is still used for the STATE.md bullet and ref hash computation.

File: scripts/persist-state-after-ship.sh [hunk 3]
```diff
@@ -299,4 +299,15 @@ generate_root_state() {
...
 
+# Garbage-collect orphaned todo-details entries after STATE.md rebuild
+DETAIL_SCRIPT=""
+SESSION_KEY="${CLAUDE_SESSION_ID:-default}"
+LINK="/tmp/.vbw-plugin-root-link-${SESSION_KEY}"
+if [ -f "${LINK}/scripts/todo-details.sh" ]; then
+  DETAIL_SCRIPT="${LINK}/scripts/todo-details.sh"
+fi
+if [ -n "$DETAIL_SCRIPT" ] && [ -f "$DETAIL_SCRIPT" ]; then
+  bash "$DETAIL_SCRIPT" gc "$OUTPUT_PATH" >/dev/null 2>&1 || true
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/persist-state-after-ship.sh]: Fixed in b778bee. Now derives `TODO_DETAILS_REGISTRY` from `OUTPUT_PATH` via `dirname` and passes it as an explicit 3rd argument to `gc`.

File: scripts/list-todos.sh [hunk 4]
```diff
@@ -264,18 +271,27 @@ main() {
       known-issue) pri_tag="[KNOWN-ISSUE] " ;;
     esac
 
-    # Strip priority prefix and date suffix for clean display
+    # Strip priority prefix, date suffix, and ref tag for clean display
     display_text="${text#\[HIGH\] }"
     display_text="${display_text#\[low\] }"
     display_text="${display_text#\[KNOWN-ISSUE\] }"
+    display_text=$(echo "$display_text" | sed 's/ *(ref:[a-f0-9]\{8\})$//')
     display_text=$(echo "$display_text" | sed 's/ *(added [0-9]\{4\}-[0-9]\{2\}-[0-9]\{2\})$//')
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/list-todos.sh]: Fixed in b778bee. Replaced both `echo` pipes with `printf '%s\n'` to handle user text containing `-n`, `-e`, or backslash sequences.

File: scripts/todo-details.sh [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+
+  # Get all keys from the registry
+  local registry_keys
+  registry_keys=$(jq -r '.items | keys[]' "$REGISTRY_PATH" 2>/dev/null || true)
+
+  # Find orphaned keys (in registry but not in STATE.md)
+  local orphaned=()
+  local key
+  for key in $registry_keys; do
+    if ! printf '%s\n' "$state_refs" | grep -qF "$key"; then
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/todo-details.sh]: Fixed in b778bee. Changed `grep -qF` to `grep -qxF` for exact-line matching — prevents substring matches where one hash is a prefix of another.

File: scripts/todo-details.sh [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+}
+
+# --- Registry validation ---
+registry_exists() {
+  [ -f "$REGISTRY_PATH" ]
+}
+
+registry_is_valid() {
+  [ -f "$REGISTRY_PATH" ] && jq -e '.schema_version and .items' "$REGISTRY_PATH" >/dev/null 2>&1
+}
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/todo-details.sh:56]: Fixed in f1de5bb. Tightened `registry_is_valid()` to explicitly check `.schema_version == 1 and (.items | type == "object")` — this rejects registries with wrong schema version or non-object items (e.g., `{"schema_version":2,"items":[]}`).

File: scripts/list-todos.sh [hunk 7]
```diff
@@ -264,18 +271,27 @@ main() {
...
-    display_text=$(echo "$display_text" | sed 's/ *(added [0-9]\{4\}-[0-9]\{2\}-[0-9]\{2\})$//')
+    display_text=$(printf '%s\n' "$display_text" | sed 's/ *(ref:[a-f0-9]\{8\})$//')
+    display_text=$(printf '%s\n' "$display_text" | sed 's/ *(added [0-9]\{4\}-[0-9]\{2\}-[0-9]\{2\})$//')
+
+    # Show 📎 indicator for todos with extended detail
+    local ref_indicator=""
+    local item_ref
+    item_ref=$(echo "$row" | base64 -d | jq -r '.ref // ""')
+    if [ -n "$item_ref" ]; then
+      ref_indicator=" 📎"
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/list-todos.sh]: Fixed in f1de5bb. Replaced `📎` with `[detail]` — plain text, no emoji, consistent with brand essentials vocabulary.

File: scripts/list-todos.sh [hunk 8]
```diff
@@ -158,14 +158,20 @@ parse_todo_line() {
     date_str="${BASH_REMATCH[1]}"
   fi
 
+  # Extract detail ref hash from (ref:<8-hex-chars>)
+  ref=""
+  if [[ "$text" =~ \(ref:([a-f0-9]{8})\) ]]; then
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/list-todos.sh]: Fixed in f1de5bb. Regex now anchored to end of string: `\(ref:([a-f0-9]{8})\)[[:space:]]*$`. This prevents false matches on hex substrings that happen to appear mid-text.

File: commands/list-todos.md [hunk 9]
```diff
@@ -63,10 +65,11 @@ allowed-tools: Read, Edit, Bash, AskUserQuestion
...
      Do NOT execute the command. STOP after displaying the suggested command.
-   - **Remove:** Remove the `line` value from the todo section in STATE.md. If no todos remain, replace with "None." Log to `## Recent Activity` with format `- {YYYY-MM-DD}: Removed todo: {text}`. Confirm: "✓ Todo removed." Run `bash "${PLUGIN_ROOT}/scripts/suggest-next.sh" list-todos` and display.
+   - **Remove:** Remove the `line` value from the todo section in STATE.md. If no todos remain, replace with "None." If the item has a non-null `ref` field, also run `bash "${PLUGIN_ROOT}/scripts/todo-details.sh" remove <ref>` to clean up the detail entry. If the detail removal fails, display "⚠ Todo removed but detail cleanup failed for ref `HASH` — run `/vbw:doctor` to clean up." rather than silently swallowing the error. Log to `## Recent Activity` with format `- {YYYY-MM-DD}: Removed todo: {text}`. Confirm: "✓ Todo removed." Run `bash "${PLUGIN_ROOT}/scripts/suggest-next.sh" list-todos` and display.
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply commands/list-todos.md]: Fixed in f1de5bb. The instruction now says to capture the JSON output and check the `status` field for `"ok"` rather than relying on exit code — since `todo-details.sh` always exits 0 (fail-open design), checking exit code would never detect failures.

File: scripts/todo-details.sh [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+  context_len=$(printf '%s' "$detail_json" | jq -r '.context // "" | length')
+  if [ "$context_len" -gt "$MAX_CONTEXT_LENGTH" ]; then
+    detail_json=$(printf '%s' "$detail_json" | jq --argjson max "$MAX_CONTEXT_LENGTH" \
+      '.context = (.context[:$max] + " (truncated)")')
+  fi
+
+  # Upsert: merge detail into items keyed by hash
+  local updated
+  updated=$(jq --arg h "$hash" --argjson d "$detail_json" \
+    '.items[$h] = $d' "$REGISTRY_PATH")
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/todo-details.sh]: Fixed in f1de5bb. Added `tostring` coercion so non-string context values (null, numbers, booleans) are safely handled, plus `2>/dev/null` and `|| context_len=0` / `|| true` guards so jq errors don't crash the script under `set -e`.

File: scripts/track-known-issues.sh [hunk 11]
```diff
@@ -818,6 +833,30 @@ promote_todos() {
...
+      local detail_json
+      detail_json=$(jq -n \
+        --arg summary "${p_test} (${p_file}): ${p_error}" \
+        --arg context "Known issue from phase ${phase_num}. Test: ${p_test}. File: ${p_file}. Error: ${p_error}. Seen ${p_times} time(s). Source: ${p_source:-unknown}. Disposition: ${p_disp:-unresolved}." \
+        --arg file "$p_file" \
+        --arg added "$today" \
+        '{summary: $summary, context: $context, files: [$file], added: $added, source: "known-issue"}')
+
+      bash "$detail_script" add "$hash" "$detail_json" "$details_path" >/dev/null 2>&1 || true
+    done <<< "$promoted_details"
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/track-known-issues.sh:864]: Fixed in f1de5bb. `full_error_msg` is now base64-encoded before storage in `promoted_details` and decoded after `IFS` parsing on the read side. This prevents multi-line error messages from breaking the `\x1f`-delimited field separator record structure.

File: commands/todo.md [hunk 12]
```diff
@@ -24,7 +24,20 @@ allowed-tools: Read, Edit
...
 2. **Parse args:** Description (non-flag text), --priority (default: normal). Format: high=`[HIGH]`, normal=plain, low=`[low]`. Append `(added {YYYY-MM-DD})`.
 3. **Add to STATE.md:** Find `## Todos` section. Replace "None." / placeholder or append after last item.
-4. **Confirm:** Display ✓ + formatted item + Next Up (/vbw:status).
+4. **Capture extended detail (conditional).** Check whether the description from step 2 contains any of these: file paths, reproduction steps, stack traces, code references, error messages, or multi-sentence design rationale. Only evaluate the current `$ARGUMENTS` text — do not scan prior conversation history.
+   - **If triggered:** The todo has rich context worth preserving for later execution.
+     1. Extract a brief one-line summary (first sentence or the user's explicit title) — this is already the `STATE.md` bullet text from step 3.
+     2. Compute a hash: `echo "<summary text>" | shasum | cut -c1-8`
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply commands/todo.md]: Fixed in 8191288. Switched to `printf '%s'` for stable hashing — avoids `echo`'s implementation-defined behavior with `-n`, `-e`, and backslash escapes.

File: scripts/list-todos.sh [hunk 13]
```diff
@@ -223,8 +230,8 @@ main() {
     num=$((num + 1))
     items_json=$(echo "$items_json" | jq --argjson n "$num" \
       --arg l "$line" --arg t "$text" --arg p "$pri" \
-      --arg d "$date_val" --arg a "$age" \
-      '. + [{num:$n, line:$l, text:$t, priority:$p, date:$d, age:$a}]')
+      --arg d "$date_val" --arg a "$age" --arg r "$ref" \
+      '. + [{num:$n, line:$l, text:$t, priority:$p, date:$d, age:$a, ref:(if $r == "" then null else $r end)}]')
   done <<< "$todo_lines"
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/list-todos.sh:235]: Fixed in 8191288. Updated the JSON schema comment on line 21 to include `"ref": "abcd1234"|null` in the items shape documentation.

File: commands/todo.md [hunk 14]
```diff
@@ -24,7 +24,20 @@ allowed-tools: Read, Edit
...
+     3. Build a JSON detail object: `{"summary": "<brief summary>", "context": "<full description, max 2000 chars>", "files": ["<any file paths mentioned>"], "added": "<YYYY-MM-DD>", "source": "user"}`
+     4. Store it — pipe JSON via heredoc to avoid shell-quoting issues with apostrophes or special characters in user text:
+        ```bash
+        bash /tmp/.vbw-plugin-root-link-${CLAUDE_SESSION_ID:-default}/scripts/todo-details.sh add <hash> - <<'DETAIL_JSON'
+        <json>
+        DETAIL_JSON
+        ```
+     5. Append `(ref:<hash>)` to the todo line in `STATE.md` (after the `(added ...)` date tag).
+   - **If not triggered** (simple one-liner with no structural context): skip — no ref tag, no detail storage. Brief bullets keep STATE.md scannable and token-efficient for context compilation. The detail file preserves context that would otherwise be lost when the todo is executed in a later session.
+5. **Confirm:** Display ✓ + formatted item + Next Up (/vbw:status). If detail was captured, also display: "📎 Extended detail saved (ref:HASH)."
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply commands/todo.md]: Fixed in 8191288. Removed 📎 and switched to plain text: `"Extended detail saved (ref:HASH)."` — consistent with brand essentials vocabulary.

File: scripts/track-known-issues.sh [hunk 15]
```diff
@@ -818,6 +836,32 @@ promote_todos() {
...
+      # Decode base64-encoded error message
+      p_error=$(printf '%s' "$p_error" | base64 -d 2>/dev/null) || p_error="(decode error)"
+
+      local detail_json
+      detail_json=$(jq -n \
+        --arg summary "${p_test} (${p_file}): ${p_error}" \
+        --arg context "Known issue from phase ${phase_num}. Test: ${p_test}. File: ${p_file}. Error: ${p_error}. Seen ${p_times} time(s). Source: ${p_source:-unknown}. Disposition: ${p_disp:-unresolved}." \
+        --arg file "$p_file" \
+        --arg added "$today" \
+        '{summary: $summary, context: $context, files: [$file], added: $added, source: "known-issue"}')
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/track-known-issues.sh:861]: Fixed in 7ef1ea7. The truncated `error_msg` (≤80 chars, matching the STATE.md bullet) is now passed as an 8th field in `promoted_details` and used for `summary`. The full decoded error stays only in the `context` field.
